Add a note about HR not being a replacement for ORM (#35777)
quarkusio/quarkus: docs/src/main/asciidoc/hibernate-reactive-panache.adoc

It makes complex mappings possible, but it does not make simple and common mappings trivial.
Hibernate Reactive with Panache focuses on making your entities trivial and fun to write in Quarkus.

[NOTE]
====
Hibernate Reactive is not a replacement for xref:hibernate-orm-panache.adoc[Hibernate ORM] or the future of Hibernate ORM.

> It is a different stack tailored for reactive use cases.

Reviewer: ```suggestion
It is a different stack tailored for reactive use cases where you need high-concurrency.
```

> Also, using RESTEasy Reactive, our default REST layer, does not mandate the use of Hibernate Reactive.

Reviewer 2: ```suggestion
Furthermore, using RESTEasy Reactive, our default REST layer, does not mandate the use of Hibernate Reactive.
```

It is perfectly valid to use RESTEasy Reactive with Hibernate ORM,

> and if you are not accustomed with the reactive paradigms, it is in most cases recommended to use Hibernate ORM.

Reviewer: ```suggestion
and if you do not need high-concurrency, or are not accustomed to the reactive paradigm, it is in most cases recommended to use Hibernate ORM.
```

> and if you do not need high-concurrency, or are not accustomed to the reactive paradigm, it is in most cases recommended to use Hibernate ORM.

Reviewer: I would remove in the most cases. I think it's clear enough.

Author: Done.
